doc/src/build/package-upgrades.md

```diff
@@ -8,6 +8,44 @@ The inability to change package objects, however, becomes a problem when conside
 
 This topic examines how to upgrade packages using the Sui Client CLI. Move developers can reference the [package module documentation](https://github.com/MystenLabs/sui/blob/main/crates/sui-framework/docs/package.md) for options on working with package upgrades on a code level.  
 
+## Upgrade considerations
+
+There are some details of the process that you should consider before upgrading your packages.
+
+For example, module initializers do not re-run with package upgrades. When you publish your initial package, Sui Move runs the `init` function you define for the package once (and only once) at the time of the publish event. Any `init` functions you might include in subsequent versions of your package are ignored.
+
+As alluded to previously, all packages on the Sui network are immutable. Because of this fact, you cannot delete old packages from the chain. As a result, there is nothing that prevents other packages from accessing the methods and types defined in the old versions of your upgraded packages. As a package developer, you must be aware of and account for this possibility.
```

Comment:
Clarifying: other packages can access the methods and types, but also users who want to keep using the old version are free to as well, by default.